engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/Interpreter.scala

```diff
@@ -154,11 +154,13 @@ private class InterpreterInternal[F[_]](listeners: Seq[ProcessListener],
     }
   }
 
-  private def interpretNext(next: Next, ctx: Context): F[List[InterpretationResult]] =
+  private def interpretNext(next: Next, ctx: Context): F[List[InterpretationResult]] = {
+    listeners.foreach(_.nodeEntered(next.id, ctx, metaData))
```

Comment:
IMO better would be leave notifing `nodeEntered` where it was before and add "additional" notify in PartRef handling with additional comment. WDYT? Also I have doubts if there are cases where we double `nodeEntered` for custom node? Once during `PartRef` handling and second during going though `case CustomNode(...) => `? Do you know it if is really possible or maybe we can have:
```
      case CustomNode(_, next) =>
        throw new IllegalStateException("Custom node should be cut off during process splitting")
```
to make sure that it is situation which will never happen

Reply:
Please note, that in `intepretNext` we invoke listener for **next**.id 
I have added test which confirms that.

Reply:
@arkadius please see the code now